chapters/arrays.tex

```diff
@@ -946,6 +947,9 @@ \section{Indexing}\label{array-indexing}\label{indexing}
 The number of dimensions of the expression is reduced by the number of scalar index arguments.
 If the number of index arguments is smaller than the number of dimensions of the array, the trailing indices will use `\lstinline!:!'.
 
+It is possible to index a general expression by enclosing it in parenthesis.
+Note that the grammar is left factored and allows indexing of \lstinline[language=grammar]!output-expression-list!, but only indexing of an expression is syntactically valid.
```

Comment:
Isn't this a semantical rule, rather than syntactical?  I suppose we could also omit the reason:
```suggestion
Note that while the subscripts are applied to a \lstinline[language=grammar]!output-expression-list! in the grammar, it is only semantically valid when the \lstinline[language=grammar]!output-expression-list! represents an expression.
```

Reply:
Well, as the grammar is currently written it is a semantic rule (due to left-factoring to make it LL(k)), but if we would switch to an LR-grammar it would be a syntactic rule.